Civi/Financeextras/Setup/Manage/AccountsReceivablePaymentMethod.php

```diff
@@ -63,4 +63,39 @@ private function getAccountsReceivablePaymentMethodId(): ?int {
     return NULL;
   }
 
+  private function getAccountIdForAccountsReceivablePaymentMethod(): int {
+    try {
+      $financialAccountId = civicrm_api3('FinancialAccount', 'getvalue', [
+        'return' => 'id',
+        'name' => self::TITLE,
+      ]);
+
+      return $financialAccountId;
+    }
+    catch (\Throwable $e) {
+      $financialAccountId = 0;
+      $assetAccountType = \CRM_Core_PseudoConstant::accountOptionValues('financial_account_type', NULL, " AND v.name = 'Asset' ");
+      if (empty($assetAccountType)) {
+        return $financialAccountId;
+      }
+
+      $financialAccounts = civicrm_api3('FinancialAccount', 'get', [
+        'sequential' => 1,
+        'financial_account_type_id' => key($assetAccountType),
```

Comment:
API v3 accept using the option value name, so you can instead do this `'financial_account_type_id' => 'Asset'` without having to prefetch it like this: `$assetAccountType = \CRM_Core_PseudoConstant::accountOptionValues('financial_account_type', NULL, " AND v.name = 'Asset' ")`

also can you use API v4 instead.

Reply:
changed